core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/upgrade-masterN.yml

```diff
@@ -0,0 +1,101 @@
+---
+- name: upgrade-master | Wait for cluster's readiness
+  include_tasks: wait.yml
+
+- name: upgrade-master | Check if /etc/kubeadm/kubeadm-config.yml exists
+  stat:
+    path: /etc/kubeadm/kubeadm-config.yml
+  changed_when: false
+  register: kubeadm_config_file
+
+- name: upgrade-master | Drain master in preparation for maintenance
+  environment:
+    KUBECONFIG: /home/{{ admin_user.name }}/.kube/config
+  shell: kubectl drain {{ inventory_hostname }} --ignore-daemonsets --delete-local-data
+  when:
+    - groups.kubernetes_node is defined
+    - groups.kubernetes_node | length > 0  # drain master only if there is at least one worker node
+
+- name: upgrade-master | Wait for cluster's readiness
+  include_tasks: wait.yml
+
+- name: upgrade-master | Upgrade, configure packages
+  block:
+    - name: upgrade-master | Hold packages for Debian family
+      include_tasks: "Debian/hold-packages.yml"
+      when: ansible_os_family == "Debian"
+
+    - name: upgrade-master | Install kubeadm
+      include_tasks: "{{ ansible_os_family }}/install-kubeadm.yml"
+
+    - name: upgrade-master | Wait for cluster's readiness
+      include_tasks: wait.yml
+
+    - name: upgrade-master | Upgrade master {{ inventory_hostname }}
+      shell: >-
```

Comment:
Any specific reason why not to use 'command' instead?

Reply:
Not really, just copy & paste. Let me work on it. :+1: